TechnicalDocuments/Proposals/BuildersToFunctions.md

```diff
@@ -0,0 +1,161 @@
+# 13. Replace Builders with Free Functions
+
+* Author(s): Rui Peres
+* Review Manager: Giorgos Tsiapaliokas
+
+## Introduction
+
+The goal of this proposal is to replace the concept of a Builder as a struct to Builder as a "namespaced" static function. This could be in the following shape:
+
+```swift
+enum Builder {}
+
+extension Builder  {
+    enum Alert {
+            static func makeAlert(for error: CoreError, dismiss: (() -> Void)?) -> UIAlertController
+            static func makeAlert(for message: String, title: String?, dismiss: (() -> Void)?) -> UIAlertController
+            static func makeAlert(for error: AlertError,
+                                  primaryAction: UIAlertAction,
+                                  secondaryAction: UIAlertAction) -> UIAlertController
+    }
+}
+```
+
+This would allow us to remove the protocol `AlertBuilder` and remove all the compliances across the codebase. From a usage point of view:
+
+```swift
+final class AppointmentListFlowController {
+    let primary: Flow
+    let secondary: Flow
+
+    init(primary: Flow, secondary: Flow) {
+        self.primary = primary
+        self.secondary = secondary
+    }
+
+    func handle(_ route: AppointmentListViewModel.Route) {
+        switch route {
+        case let .showAlert(error):
+            Builder.Alert.makeAlert(for: error, dismiss: secondary.dismiss)
+                |> secondary.present
+        }
+    }
+}
+```
+
+I will provide further examples in the proposal
+
+## Motivation
+
+There are a couple of reasons for this change:
+
+1. It was decided that DI via `Current` would happen at the builders level. This means that the requirement to inject dependencies via initializer, in builders, cease to exist. This would remove: protocols, clean FlowControllers APIs and dissolve the concept of child builders. 
```

Comment:
```suggestion
1. It was decided that DI via `Current` would happen at the builder level. This means that the requirement to inject dependencies via the initializer, in builders, ceases to exist. This would: remove protocols, clean FlowController's APIs and dissolve the concept of child builders. 
```